packages/typescript/src/index.ts

```diff
@@ -158,13 +158,17 @@ export default function typescript(options: RollupTypescriptOptions = {}): Plugi
             (parsedOptions.options.declaration
               ? parsedOptions.options.declarationDir || parsedOptions.options.outDir
               : null);
-          const cwd = normalizePath(process.cwd());
+          const outputDir = normalizePath(
+            outputOptions.dir || !outputOptions.file
+              ? process.cwd()
```

Comment:
Shouldn't the output dir be `outputOptions.dir`?

Reply:
If `outputOptions.dir` is set, then *it is used*.

If `outputOptions.dir` is **not** set, then - if `outputOptions.file` is set, the base directory of it is used, if not then the current working dir is used.

Reply:
I see. Is the execution like this?
```js
let dir = outputOptions.dir;

if (!dir) {
  dir = !outputOptions.file ? process.cwd() : path.dirname(outputOptions.file)
}

const outputDir = normalizePath(dir);
```

If so, it would be nice to have parenthesis around the latter `!outputOptions.file ? process.cwd() : path.dirname(outputOptions.file)` part so the code execution is clearer. If prettier removes it, then you can ignore this comment.

as a nit, I would prefer to write `outputOptions.file ? path.dirname(outputOptions.file) : process.cwd()` rather than `!outputOptions.file ? process.cwd() : path.dirname(outputOptions.file)`